README.md

```diff
@@ -73,8 +73,11 @@ Even though it is strongly preferred to rely on ECC P521 and SHA-512 where possi
 6. EnrollZ service obtains the oIAK and oIDevID certs from the CA and calls the device's `RotateOIakCert` API to persist the oIAK and oIDevID certs on the control card.
 7. The switch verifies that the IAK pub key in oIAK cert matches the one in IAK cert and that IDevID pub key in oIDevID cert matches the one in IDevID cert.
 8. The switch stores oIAK and oIDevID certs in non-volatile memory and will present them in the TPM attestation `attestz` workflow.
-9. The switch must update its default SSL profile to rely on the owner IDevID cert instead of IDevID cert.
-   - *Note: This implies that after successful enrollment the switch must force all its gRPC servers/services (such as `attestz` and `certz`) to respect the updated SSL profile relying on oIDevID cert.*
+9. The switch must use the profile created during bootz which provided the trust bundle. This will rotate the profiles cert to be the provided Owner IDevID cert and
```

Comment:
During Bootz, the Bootstrap response provides the "gNSI.Certz.UploadRequest" artifact, which was initially intended to the "gNXI" profile. With recent changes, there meant to be a "system-default-profile" which is meant to be immutable, has IDevID cert, and is expected to be enabled by default. So when Bootstrap server sends the entities as part of the UploadRequest, what is the expectation in terms of creating a new SSL profile? Is the system meant to create a brand new profile with a well-known keyword? Or should the Bootz spec add an "AddProfileRequest" request?

Reply:
Currently Bootz only includes a Certz.UploadRequest data structure, which does not specify the profile name. Could you clarify how bootz specified the profile name? Are there any changes planned to support Bootz specifying the profile name?

Reply:
There are two possible approaches:
1) the profile can be defined as a part of the bootconfig, or
2) the profile can be defined as a gnsi artifact in the bootstrapResponse, in that case, as @pranav-jnpr indicates, the bootz.proto should be updated to allow specifying a full profile, and not only certificates data.

Reply:
I think latter makes the most sense

Reply:
https://github.com/openconfig/bootz/pull/130

suggestion for change to bootz

Reply:
LGTM, even for other deployment options listed in the bootz spec it makes sense to add a profile-id here.

The alternative way I was thinking was adding -
1. AddProfileRequest
2. Replacing Certz.UploadRequest with Certz.RotateRequest - since this structure has the ssl-profile-id, and also includes "version" and "Created-on" fields

With changes in bootz pull/130 the new message field requires the system to co-relate the profile-id, treat it as an AddProfileRequest, and use the entities in UploadRequest to rotate into the profile-id. Which also seems doable, just requires new logic.

Reply:
i think it is clearer on intent as fith add profile request still would need to bind the profile to entities so you saying create this list - then add these things to the previous items - seems clearer to just add artifacts and create profile together since it actually makes it less error prone 

i am not following your #2?

there is no certz.RotateRequest that i see
and Entity contain the version / created-on which is a repeated inside UploadRequest?

if you mean RotateCertificateRequest, we didn't include that as it includes a oneof that defines the "stream protocol" which we wouldn't include in bootz handling. basically we would never expect bootz to see a finalize message in the data

Reply:
Thanks for explaining the reasoning behind not using the "RotateCertificateRequest". I am now thinking how we populate the "version" and "created-on" fields for those entities uploaded by Bootz, we would need to stream them through telemetry. 

For the profile-id part, what has been suggested in pull/130 seems doable and LGTM.